crates/sui/src/keytool.rs

```diff
@@ -351,6 +371,83 @@ impl KeyToolCommand {
                 }
             }
+            KeyToolCommand::SignKMS {
+                data,
+                keyid,
+                intent,
+            } => {
+                // Currently only supports secp256k1 keys
+                println!("Raw tx_bytes to execute: {}", data);
+                let intent = intent.unwrap_or_else(Intent::sui_transaction);
+                println!("Intent: {:?}", intent);
+                let msg: TransactionData =
+                    bcs::from_bytes(&Base64::decode(&data).map_err(|e| {
+                        anyhow!("Cannot deserialize data as TransactionData {:?}", e)
+                    })?)?;
+                let intent_msg = IntentMessage::new(intent, msg);
+                println!(
+                    "Raw intent message: {:?}",
+                    Base64::encode(bcs::to_bytes(&intent_msg)?)
+                );
+                let mut hasher = DefaultHash::default();
+                hasher.update(bcs::to_bytes(&intent_msg)?);
+                let digest = hasher.finalize().digest;
+                println!("Digest to sign: {:?}", Base64::encode(digest));
+
+                // Set up the KMS client, expect to use env vars or whatever the AWS
+                // SDK prefers.
+                let region = Region::default();
+                let kms = KmsClient::new(region);
+
+                // Get Pub key from keyid. This is in DER Format from AWS
+                let req = GetPublicKeyRequest {
+                    grant_tokens: None,
+                    key_id: keyid.to_string(),
+                };
+                let pubkey_resp = kms
+                    .get_public_key(req)
+                    .await
+                    .map_err(|e| anyhow!("No Public key Found: {:?}", e))?;
+                let public_key: &[u8] = &pubkey_resp.public_key.unwrap_or_default();
+
+                // Parse the PEM-encoded public key and extract the raw bytes
+                // Compresses into 33 bytes
+                let secp_pk = Secp256k1PublicKey::from_uncompressed(public_key);
+                let pkey_compact = secp_pk.as_bytes();
+                // Generates Corresponding Sui Address from public key
+                println!(
+                    "Address For Corresponding KMS Key: {}",
+                    Into::<SuiAddress>::into(&secp_pk),
+                );
+
+                // Construct the signing request
+                let request = SignRequest {
+                    key_id: keyid.to_string(),
+                    message: digest.to_vec().into(),
+                    message_type: Some("RAW".to_string()),
+                    signing_algorithm: "ECDSA_SHA_256".to_string(),
+                    ..Default::default()
+                };
+                // Sign the message, normalize the signature and then compacts it
+                let response = kms.sign(request).await?;
+                let sig_bytes_der = response.signature.map(|b| b.to_vec()).unwrap_or_default();
```

Comment:
same here, why do you care about the default rather than just handle it as error?